src/resources/js/v2/tickets-modal.js

```diff
@@ -0,0 +1,362 @@
+ *
+ * @since TBD
+ *
+ * @param  {PlainObject} $   jQuery
+ * @param  {PlainObject} obj tribe.tickets.modal
+ *
+ * @return {void}
+ */
+( function( $, obj ) {
+	'use strict';
+	const $document = $( document );
+
+	/**
+	 * Selectors used for configuration and setup.
+	 *
+	 * @since TBD
+	 *
+	 * @type {PlainObject}
+
+	obj.selectors = {
+		hiddenElement: '.tribe-common-a11y-hidden',
+	};*/
+
+	/*
+	 * AR Cart Modal Selectors.
+	 *
+	 * Note: some of these have the modal class as well, as the js can
+	 * pick up the class from elsewhere in the DOM and grab the wrong data.
+	 *
+	 * @since TBD
+	 */
+	obj.selectors = {
+		cartForm: '.tribe-modal__wrapper--ar #tribe-modal__cart',
+		container: '.tribe-modal__wrapper--ar',
+		form: '#tribe-tickets__modal-form',
+		itemRemove: '.tribe-tickets__item__remove',
+		itemTotal: '.tribe-tickets__item__total .tribe-amount',
+		loader: '.tribe-tickets-loader__modal',
+		metaField: '.tribe-tickets__form-field-input', //'.ticket-meta',
+		metaForm: '.tribe-modal__wrapper--ar #tribe-modal__attendee_registration',
+		metaItem: '.tribe-ticket',
+		submit: '.tribe-block__tickets__item__attendee__fields__footer_submit',
+	};
+
+	/**
+	 * Appends AR fields when modal cart quantities are changed.
+	 *
+	 * @since TBD
+	 *
+	 * @param {object} $form - The form we are updating.
+	 */
+
+	obj.appendARFields = function( $form ) {
+		$form.find( tribe.tickets.block.selectors.item ).each(
+			function() {
+				const $cartItem = $( this );
+
+				if ( $cartItem.is( ':visible' ) ) {
+					const ticketID = $cartItem.closest( tribe.tickets.block.selectors.item ).data( 'ticket-id' );
+					const $ticketContainer = $( obj.selectors.metaForm ).find( '.tribe-tickets__item__attendee__fields__container[data-ticket-id="' + ticketID + '"]' );
+
+					// Ticket does not have meta - no need to jump through hoops ( and throw errors ).
+					if ( ! $ticketContainer.length ) {
+						return;
+					}
+
+					const $existing = $ticketContainer.find( obj.selectors.metaItem );
+					const qty = tribe.tickets.block.getQty( $cartItem );
+
+					if ( 0 >= qty ) {
+						$ticketContainer.removeClass( 'tribe-tickets--has-tickets' );
+						$ticketContainer.find( obj.selectors.metaItem ).remove();
+
+						return;
+					}
+
+					if ( $existing.length > qty ) {
+						const removeCount = $existing.length - qty;
+						$ticketContainer.find( '.tribe-ticket:nth-last-child( -n+' + removeCount + ' )' ).remove();
+
+					} else if ( $existing.length < qty ) {
+
+						const ticketTemplate = window.wp.template( 'tribe-registration--' + ticketID );
+						const counter = 0 < $existing.length ? $existing.length + 1 : 1;
+
+						$ticketContainer.addClass( 'tribe-tickets--has-tickets' );
+
+						for ( let i = counter; i <= qty; i++ ) {
+							const data = { attendee_id: i };
+
+							$ticketContainer.append( ticketTemplate( data ) );
+							obj.maybeHydrateAttendeeBlockFromLocal( $existing.length );
+						}
+					}
+				}
+			}
```

Comment:
Separate method.

Reply:
This is not ready for review, pushed so Scott could know where we were at.